apps/api/routes/query.py

```
@@ -0,0 +1,76 @@
+            rerank_top_n=request.rerank_top_n,
+            source_types=request.source_types,
+            after=request.after
+        )
+
+        if not result:
+            raise HTTPException(status_code=500, detail="Query execution failed")
+
+        return QueryResponse(**result)
+
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)